### ISSUE TITLE
Update SA1302 (InterfaceNamesMustBeginWithI) to handle conflicting names

### ISSUE THREAD

**Issue by sharwell** [DotNetAnalyzers/StyleCopAnalyzers#436]
The following code results in an SA1302 warning. If the code fix is applied, it introduces a new compiler error.

```csharp
public interface Foo { }
public interface IFoo { }
public class Bar : Foo { }
```

We first need to decide how we want to handle this case (and similar), followed by implementation and tests.

**Comment by macpak** [comment]
First of all we need to check if a type with the new name exists in the same namespace. If it's in a different namespace - it's safe to apply the fix. If the type aleady exists in the namespace, maybe we should append a number to the name, to make it unique  (of course, if we have Foo interface, and IFoo and IFoo2  already exist, we should rename it to IFoo3).

**Comment by sharwell** [comment]
We *could* do that. First I'm curious if there is a way to check with `Renamer` to see if the rename operation introduced new errors, and cancel the code fix if it did.

**Comment by macpak** [comment]
Isn't it done in #443 ?

**Comment by sharwell** [comment]
No, that pull request makes improvements based on the feedback in dotnet/roslyn#197, but this particular issue did not behave in the manner expected by that team. Hopefully the next release of the code fix framework allows us to notify the user of a conflict prior to applying it.
